### PR TITLE
Out-of-band initial ROC

### PR DESCRIPTION
**Pull request by algesten**

### CODE REVIEW

File: src/streams/send.rs [hunk 1]
```diff
@@ -254,7 +261,15 @@ impl StreamTx {
         nackable: bool,
         payload: Vec<u8>,
     ) -> Result<(), RtcError> {
-        //
+        if self.start_roc.is_none() {
+            // This is the first ever write_rtp. Remember the starting ROC.
+            self.start_roc = Some(*seq_no >> 16);
+        }
+
+        // SeqNo must start from 0.
```
Comment (algesten) [inline src/streams/send.rs:269]: Wrong comment

File: src/streams/receive.rs [hunk 2]
```diff
@@ -597,6 +615,23 @@ impl StreamRx {
         self.rtx = Some(rtx);
         self.register_rtx = None;
     }
+
+    /// Reset the current rollover counter (ROC).
+    ///
+    /// This is used in scenarios where we use a single sequence number across all
+    /// receivers of the same stream (as opposed to a sequence number unique per peer).
+    ///
+    /// > Receivers joining an on-going session MUST be given the
```
Comment (algesten) [inline src/streams/receive.rs:624]: Link to RFC

File: src/rtp/id.rs [hunk 3]
```diff
@@ -189,6 +202,23 @@ impl SeqNo {
...
+    /// // More than 2^16, thus rolled over.
+    /// let seq_no: SeqNo = 95_000.into();
+    ///
+    /// assert_eq!(seq_no.roc(), 1);
+    ///
+    /// // Is the same as shifting 16 bits.
+    /// assert_eq!(seq_no.roc(), 95_000 >> 16);
+    /// ```
+    #[inline(always)]
+    pub fn roc(&self) -> u64 {
```
Comment (k0nserv) [inline src/rtp/id.rs:219]: Nice, was just thinking that this method would be useful. Can you adopt in `src/rtp/srtp.rs`? too?

File: src/rtp/id.rs [hunk 4]
```diff
@@ -161,6 +161,19 @@ num_id!(SessionId, u64);
...
 /// str0m keeps track of the roll overs and converts the `u16` to `u64` in this `SeqNo`.
+///
+/// To get the RTP u16 value from a `SeqNo`, deref and cast it.
+///
+/// ```
+/// # use str0m::rtp::SeqNo;
+/// let seq_no: SeqNo = 65_537.into();
+///
+/// // Discard upper 48 bits to get RTP u16.
+/// let rtp_seq_no = *seq_no as u16;
```
Comment (k0nserv) [inline src/rtp/id.rs:172]: For symmetry should we have a method that does this too?

File: src/streams/receive.rs [hunk 5]
```diff
@@ -597,6 +615,23 @@ impl StreamRx {
...
+    /// > current ROC value using out-of-band signaling such as key-management
+    /// > signaling.  Furthermore, the receiver SHALL initialize s_l to the RTP
+    /// > sequence number (SEQ) of the first observed SRTP packet (unless the
+    /// > initial value is provided by out of band signaling such as key
+    /// > management).
+    pub fn reset_roc(&mut self, roc: u64) {
```
Comment (k0nserv) [inline src/streams/receive.rs:630]: Don't we need to be able to set the RTX ROC too? If a session has been going for long enough the RTX ROC might also be non-zero when the receiver joins

File: src/rtp/id.rs (hunk 4 above)
Comment (algesten) [reply src/rtp/id.rs:172]: Give me a name. I tried to come up with one :)

File: src/streams/receive.rs (hunk 5 above)
Comment (algesten) [reply src/streams/receive.rs:630]: The RTX is peer-peer (not entire session), and it's not possible for the user to chose a value for it.

File: src/rtp/id.rs (hunk 3 above)
Comment (algesten) [reply src/rtp/id.rs:219]: Sure

File: src/rtp/id.rs (hunk 4 above)
Comment (k0nserv) [reply src/rtp/id.rs:172]: What, you don't like `seq_no.seq_no()`? 😂 

`as_u16`, `truncated_u16`, `truncated`, `without_roc`, `omitting_roc`

File: src/streams/receive.rs (hunk 5 above)
Comment (k0nserv) [reply src/streams/receive.rs:630]: Cool 👍🏼

**Review by k0nserv — APPROVED** [review]

File: src/rtp/id.rs (hunk 3 above)
Comment (k0nserv) [reply src/rtp/id.rs:239]: Ah, of course.

**Pull request merged**